### ISSUE TITLE
MySQL data import is taking too long...

### ISSUE THREAD

**Issue by MustafaKarabulut** [cBioPortal/datahub#1214]
Hi,

I am trying to import MySQL dump that can be downloaded from the below link into my local database instance. 

http://download.cbioportal.org/mysql-snapshots/mysql-snapshots-toc.html

However, it takes too long. After a few days of waiting for the mysql import process, it just imported less than half of the tables. I can see it is about the size of tables, some of which have over 10 million rows and inserting batch of rows into an already large table requires too much time.

Is there way to speed up this process like getting raw MySQL files from somewhere and copy them directly on the machine? Or maybe I am missing some optimization tips about mysql import process?

I am doing mysql import with the below command:

`mysql -uroot -hlocalhost -pxxx cbioportal < public-portal-dump.sql`

I am open to all kind of suggestions.

Thank you

**Comment by jjgao** [comment]
Hi, there are various ways to improve, e.g. https://stackoverflow.com/questions/29643714/improve-speed-of-mysql-import and https://gabi.dev/2016/05/17/fast-data-import-trick/. Please let us know if you find any solution.

**Comment by MustafaKarabulut** [comment]
Hi,

Thank you for the reply and links for information.

Before posting here, I checked several ways to improve mysql import process, however, all things I tried helped just in a minor way. MySQL dump file is a large file which is approximately 40GBs in size, it is reasonable to expect it to take some time even with the improvements to speed it up.

I was importing it just for exploration purposes; I am trying to do a research about cancer-gene mutations relationships and first would like to see what kind of data is inside the database.

So my solution was to remove some of the really big tables off the dump like `genetic_alteration` (which occupies almost 90% of the dump in size). This way, I could finish the dump process at least in a few hours.
